### PR TITLE
enhancement/subject-search-page-and-card-mui

### PR DESCRIPTION
**Pull request by Yevheniy2020**

### CODE REVIEW

File: src/components/common/ui/cards/subject-card/SubjectCard.styles.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { SxProps } from '@mui/material/styles';
+import { Theme } from '@mui/system';
+
+import theme from '@/styles/theme';
+export const subjectCard = (disabled: boolean): SxProps<Theme> => ({
```
Comment (andrewvitrenko) [inline src/components/common/ui/cards/subject-card/SubjectCard.styles.ts:5]: add empty line above

File: src/components/common/ui/cards/subject-card/SubjectCard.styles.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  display: 'flex',
+  flexDirection: 'column',
+  gap: '8px',
+  ...(!disabled && {
+    cursor: 'pointer',
+    '&:focus': {
+      borderColor: theme.palette.primary[300],
+    },
+    '&:active': {
+      // borderColor: theme.palette.primary['logo'],
```
Comment (andrewvitrenko) [inline src/components/common/ui/cards/subject-card/SubjectCard.styles.ts:22]: remove comment

File: src/components/common/ui/cards/subject-card/SubjectCard.tsx [hunk 3]
```diff
@@ -1,31 +1,25 @@
...
 
-import styles from '@/components/common/ui/cards/subject-card/SubjectCard.module.scss';
-import { DivProps } from '@/components/common/ui/cards/types';
-
-type SubjectCardProps = {
+import * as styles from './SubjectCard.styles';
+interface SubjectCardProps {
   name: string;
   details?: string;
   rating?: number;
```
Comment (andrewvitrenko) [inline src/components/common/ui/cards/subject-card/SubjectCard.tsx:8]: where this prop is going to be used?

File: src/components/pages/search-pages/SearchPage.styles.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+};
+
+export const form: SxProps<Theme> = {
+  display: 'grid',
+  gridTemplateRows: '1fr',
+  gap: '12px',
+  marginBottom: '34px',
+  alignItems: 'center',
+  gridTemplateColumns: {
+    tablet: '35% 45% 10%',
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/SearchPage.styles.ts:29]: need to change it to usual numbers. and better use `fr`

File: src/components/pages/search-pages/SearchPage.styles.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+export const dropdownOne: SxProps<Theme> = {
+  width: '100%',
+};
+
+export const dropdownTwo: SxProps<Theme> = {
+  width: '100%',
+};
+
+export const subjectTitle: SxProps<Theme> = {
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/SearchPage.styles.ts:50]: remove this

File: src/components/pages/search-pages/SearchPage.styles.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    tablet: 'center',
+    desktop: 'auto',
+  },
+};
+
+export const collapseBtn: SxProps<Theme> = {
+  display: 'none',
+};
+
+export const dropdownOne: SxProps<Theme> = {
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/SearchPage.styles.ts:42]: leave only one dropdown

File: src/components/pages/search-pages/SearchPage.styles.ts [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    tablet: '35% 45% 10%',
+    desktop: '35% 15% 15% 5%',
+  },
+  justifyContent: {
+    tablet: 'center',
+    desktop: 'auto',
+  },
+};
+
+export const collapseBtn: SxProps<Theme> = {
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/SearchPage.styles.ts:38]: if there is only `display: none`, why don't you remove this button from page?

File: src/components/pages/search-pages/SearchPage.styles.ts [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+  // @include h6--bold;
+  // }
+};
+
+export const subjectTeacherSearchList: SxProps<Theme> = {
+  marginTop: '40px',
+  marginBottom: '16px',
+  display: 'grid',
+  gridTemplateColumns: 'repeat(4, 1fr)',
+  rowGap: '4rem',
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/SearchPage.styles.ts:63]: we use pixels on this project

File: src/components/pages/search-pages/SearchPage.styles.ts [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+  paddingBottom: '30px',
+  width: '100%',
+  display: 'flex',
+  justifyContent: 'center',
+};
+
+export const collapseIcon: SxProps<Theme> = {
+  height: '48px',
+  aspectRatio: '1/1',
+  // alignSelf: 'flex-start',
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/SearchPage.styles.ts:80]: remove comment

File: src/components/pages/search-pages/SearchPage.styles.ts [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+  alignSelf: 'center',
+  display: {
+    tablet: 'block !important',
+    desktop: 'grid',
+  },
+};
+
+export const sortIcon: SxProps<Theme> = {
+  aspectRatio: '1/1',
+  // alignSelf: 'flex-start',
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/SearchPage.styles.ts:90]: remove comment

File: src/components/pages/search-pages/SearchPage.styles.ts [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+  paddingTop: '30px',
+  paddingBottom: '30px',
+  width: '100%',
+  display: 'flex',
+  justifyContent: 'center',
+};
+
+export const collapseIcon: SxProps<Theme> = {
+  height: '48px',
+  aspectRatio: '1/1',
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/SearchPage.styles.ts:79]: if you're adding static height you can also add static width

File: src/components/pages/search-pages/subject-search/components/SubjectSearchList.tsx [hunk 12]
```diff
@@ -24,10 +24,10 @@ export const SubjectSearchList = ({ subjects }: GetListOfSubjectsResponse) => {
...
 
   return (
-    <ul className={styles['subject-search-list']}>
+    <Box component="ul" sx={styles.subjectList}>
       {subjects &&
         subjects.map(subject => (
-          <li key={subject.id}>
+          <Box component="li" key={subject.id}>
             <SubjectCard
               onClick={() => redirect(subject.id)}
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-search/components/SubjectSearchList.tsx:32]:
```js
const redirect = useCallback((subjectId: string) => () => {
  void router.push(`/subjects/${subjectId}/teachers`)
}, []);
```

File: src/components/pages/search-pages/subject-search/components/SubjectSearchList.styles.ts [hunk 13]
```diff
@@ -0,0 +1,18 @@
+import { SxProps } from '@mui/material/styles';
+import { Theme } from '@mui/system';
+
+export const subjectList: SxProps<Theme> = {
+  marginBottom: '16px',
+  display: 'block',
+  flexDirection: 'row',
+  li: {
+    marginBottom: '16px',
+    breakInside: 'avoid',
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-search/components/SubjectSearchList.styles.ts:10]: why do we need this?

File: src/components/pages/search-pages/subject-search/components/SubjectSearchList.styles.ts [hunk 14]
```diff
@@ -0,0 +1,18 @@
+import { SxProps } from '@mui/material/styles';
+import { Theme } from '@mui/system';
+
+export const subjectList: SxProps<Theme> = {
+  marginBottom: '16px',
+  display: 'block',
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-search/components/SubjectSearchList.styles.ts:6]: `display: block` and `flexDirection: row`))

File: src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx [hunk 15]
```diff
@@ -52,9 +54,9 @@ const SubjectSearchPage = () => {
   }, [queryObj, curPage, refetch]);
 
   return (
-    <div className={styles['layout']}>
+    <Box sx={stylesSearch.layout}>
       {/*TODO move inline styles when refactor*/}
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx:58]: remove `todo`

File: src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx [hunk 16]
```diff
@@ -65,20 +67,20 @@ const SubjectSearchPage = () => {
...
+          </Box>
         ))}
       {data?.subjects?.length === (curPage + 1) * pageSize && (
         <Button
-          className={styles['load-btn']}
+          sx={stylesPage.loadBtn}
           text="Завантажити ще"
           variant={ButtonVariant.FILLED}
           color={ButtonColor.SECONDARY}
           onClick={() => setCurPage(pr => pr + 1)}
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx:80]: move this to the separate callback and rename `pr` to `prev`

File: src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx [hunk 17]
```diff
@@ -1,22 +1,24 @@
...
 
 import { SubjectInitialValues } from '../search-form/constants';
 import { SearchForm } from '../search-form/SearchForm';
+import * as stylesSearch from '../SearchPage.styles';
 
-import { SubjectSearchList } from './SubjectSearchList';
-
-import styles from '../SearchPage.module.scss';
+import { SubjectSearchList } from './components/SubjectSearchList';
+import * as stylesPage from './SubjectSearchPage.styles';
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx:21]: rename to `styles`

File: src/components/pages/search-pages/SearchPage.styles.ts [hunk 18]
```diff
@@ -0,0 +1,108 @@
...
+
+export const pageLoader: SxProps<Theme> = {
+  paddingTop: '30px',
+  paddingBottom: '30px',
+  width: '100%',
+  display: 'flex',
+  justifyContent: 'center',
+};
+
+export const collapseIcon: SxProps<Theme> = {
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/SearchPage.styles.ts:77]: I don't see where you used these styles

File: src/components/pages/search-pages/SearchPage.styles.ts [hunk 19]
```diff
@@ -0,0 +1,108 @@
...
+  aspectRatio: '1/1',
+  // alignSelf: 'flex-start',
+  alignSelf: 'center',
+  display: {
+    tablet: 'block !important',
+    desktop: 'grid',
+  },
+};
+
+export const sortIcon: SxProps<Theme> = {
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/SearchPage.styles.ts:88]: and I don't see usage of these also

File: src/components/pages/search-pages/subject-search/components/SubjectSearchList.tsx [hunk 20]
```diff
@@ -24,10 +24,10 @@ export const SubjectSearchList = ({ subjects }: GetListOfSubjectsResponse) => {
   };
 
   return (
-    <ul className={styles['subject-search-list']}>
+    <Box component="ul" sx={styles.subjectList}>
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-search/components/SubjectSearchList.tsx:27]: mui has elements for list, you can check them

**Review by andrewvitrenko — CHANGES_REQUESTED** [review]

File: src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx (hunk 16 above)
Comment (Yevheniy2020) [reply src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx:80]: should I use hook useCallback here?

File: src/components/pages/search-pages/subject-search/components/SubjectSearchList.tsx (hunk 20 above)
Comment (Yevheniy2020) [reply src/components/pages/search-pages/subject-search/components/SubjectSearchList.tsx:27]: I can't find(

File: src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx (hunk 16 above)
Comment (Yevheniy2020) [reply src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx:80]: should I use hook useCallback here?

File: src/components/common/ui/cards/subject-card/SubjectCard.tsx (hunk 3 above)
Comment (andrewvitrenko) [reply src/components/common/ui/cards/subject-card/SubjectCard.tsx:8]: rating

File: src/components/pages/search-pages/SearchPage.styles.ts (hunk 7 above)
Comment (Yevheniy2020) [reply src/components/pages/search-pages/SearchPage.styles.ts:38]: image

File: src/components/pages/search-pages/subject-search/components/SubjectSearchList.tsx [hunk 21]
```diff
@@ -19,17 +20,27 @@ export const SubjectSearchList = ({ subjects }: GetListOfSubjectsResponse) => {
...
-  const redirect = (subjectId: string) => {
-    void router.push(`/subjects/${subjectId}/teachers`);
+  const redirect = useCallback(
+    (subjectId: string) => () => {
+      void router.push(`/subjects/${subjectId}/teachers`);
+    },
+    [],
+  );
+
+  const breakpoints = {
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-search/components/SubjectSearchList.tsx:30]: it is better to move this to `constants` folder

File: src/components/pages/search-pages/subject-teacher-search/SearchTeacherPage.tsx [hunk 22]
```diff
@@ -21,7 +23,7 @@ const breadcrumbs: Breadcrumb[] = [
   },
 ];
 
-const SubjectTeacherPage = () => {
+const SearchTeacherPage = () => {
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-teacher-search/SearchTeacherPage.tsx:26]: add `FC` type

**Review by andrewvitrenko — CHANGES_REQUESTED** [review]

File: src/components/pages/search-pages/subject-search/SubjectSearchPage.styles.ts [hunk 23]
```diff
@@ -0,0 +1,37 @@
...
+  alignSelf: 'center',
+};
+
+export const layout: SxProps<Theme> = {
+  width: '100%',
+  display: 'flex',
+  flexDirection: 'column',
+  padding: {
+    tablet: '12px 16px',
+    desktop: '0px 80px 32px 80px',
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-search/SubjectSearchPage.styles.ts:19]: you can remove the last `80px`

File: src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx [hunk 24]
```diff
@@ -1,22 +1,24 @@
...
-} from '@/components/common/ui/button/Button';
+} from '@/components/common/ui/button-mui/types';
 import Progress from '@/components/common/ui/progress-mui';
 import { SearchFormProps } from '@/components/pages/search-pages/search-form/SearchForm';
 import SubjectsAPI from '@/lib/api/subject/SubjectAPI';
 import { GetListOfSubjectsResponse } from '@/lib/api/subject/types/GetListOfSubjectsResponse';
 
 import { SubjectInitialValues } from '../search-form/constants';
 import { SearchForm } from '../search-form/SearchForm';
+import * as stylesSearch from '../SearchPage.styles';
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx:18]: where do you use that?

**Review by andrewvitrenko — CHANGES_REQUESTED** [review]

**Review by andrewvitrenko — APPROVED** [review]

**Pull request merged**